src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -15,34 +16,31 @@ public GildedRose(Item[] items) {
     }
 
     public void updateQuality() {
+        int id = 1;
         for (Item item : items) {
-            Ageable ageable = inventory.get(item);
+            Ageable ageable = inventory.get(String.valueOf(id));
             ageable.age();
+            id++;
         }
     }
 
-    private HashMap<Item, Ageable> populateInventory() {
-        HashMap<Item, Ageable> itemInventory = new HashMap<>();
+    public Map<String,Ageable> getInventory() {
+        return this.inventory;
+    }
+
+    private HashMap<String, Ageable> populateInventory() {
```

Comment:
This looks very similar to the `populateHashMapDB` method in the repository class. What are the difference?

Reply:
See my comment above about how the Repository could take a list of items as a dependency in its constructor. That would clean up this duplication of knowledge and configuration.

Reply:
Question: GildedRose takes a list of Items as the inventory, what do we do with those?

Reply:
GildedRose is now the runner/top level of the application, so the new GildedRose is the only place the inventory population exists. We populate the database with the corresponding items ID as the key and its corresponding ageable representation as the value. This DB is then injected into the Repository as @jramnani was talking about in the comment below!

Reply:
This can be seen in [this commit here](https://github.com/jramnani/GildedRoseJavaLLC/pull/24/commits/0870d6cc7439ad96d3eb4d8b41f283f221c5511d)!